internal/mutators/podtemplatespec_test.go

```diff
@@ -0,0 +1,362 @@
+package mutators
+
+import (
+	"testing"
+
+	"github.com/kanopy-platform/hedgetrimmer/internal/limitrange"
+	"github.com/kanopy-platform/hedgetrimmer/internal/quantity"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestApplyResourceRequirements(t *testing.T) {
+	t.Parallel()
+
+	limitRangeItem := corev1.LimitRangeItem{
+		Type:           corev1.LimitTypeContainer,
+		DefaultRequest: corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("50Mi")},
+		Default:        corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("64Mi")},
+	}
+
+	tests := []struct {
+		inputPts  corev1.PodTemplateSpec
+		inputLri  corev1.LimitRangeItem
+		want      corev1.PodTemplateSpec
+		wantError bool
+		msg       string
+	}{
+		{
+			inputPts: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					InitContainers: []corev1.Container{
+						{
+							Name:      "init-container-1",
+							Resources: corev1.ResourceRequirements{}, // empty requests and limits
```

Comment:
nit: comments are probably easier to parse at the top of each test case.

Reply:
I assume "empty" and "no <resource> limit is the same.

For test setup, I'd opt for better function names than comments that clearly denote what is being provided.

```suggestion
							Resources: noLimtResourceRequests(),
```